### PR TITLE
feat: create related prompts module

### PR DESCRIPTION
**Pull request by victorcg88**
# Pull request template
 
The Related Prompts module. Includes:

- The Related Prompts store including query, status, and extra params.
- The RelatedPrompt component. Including three slots: `header`, `next queries`, and `query-preview`.
- The RelatedPromptList component. Same functionality as the NextQueriesList component. (Injects related prompts in the results list)

## Motivation and context

- [ ] Dependencies. If any, specify:
- [ ] Open issue. If applicable, link:

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [ ] `Main`
- [X] Other. Specify: https://github.com/empathyco/x/tree/feat/add-related-prompts-adapter

## How has this been tested?

Tests performed according to [testing guidelines](./contributing/tests.md): 

## Checklist:

- [x] My code follows the **[style guidelines](./CONTRIBUTING.md#style-guides)** of this project.
- [x] I have performed a **self-review** on my own code.
- [x] I have **commented** my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the **[documentation](./CONTRIBUTING.md#documentation-style-guide)**.
- [x] My changes generate **no new warnings**.
- [x] I have added **tests** that prove my fix is effective or that my feature works.
- [x] New and existing **unit tests pass locally** with my changes.

### CODE REVIEW

File: packages/x-components/src/x-modules/related-prompts/components/related-prompt.vue [hunk 1]
```diff
@@ -0,0 +1,80 @@
+<template>
+  <div
+    class="x-related-prompt x-bg-neutral-10 x-p-24 x-flex x-flex-col x-gap-16"
```
Comment (diegopf) [inline packages/x-components/src/x-modules/related-prompts/components/related-prompt.vue:3]: I'm not sure about the styling. Usually we don't add those styles to the x-component and we leave it to the consumer to style it as desired.

File: packages/x-components/src/x-modules/related-prompts/components/related-prompt.vue [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        {{ relatedPrompt.suggestionText }}
+      </slot>
+      <slot name="next-queries" :nextQueries="relatedPrompt.nextQueries">
+        <SlidingPanel :resetOnContentChange="false">
+          <div class="x-flex x-gap-8 x-pr-8">
+            <button
+              v-for="(nextQuery, index) in relatedPrompt.nextQueries"
+              :key="index"
+              @click="onClick(nextQuery)"
+              class="x-button x-button-lead x-button-sm x-button-outlined x-rounded-sm x-border-lead-50 x-text-neutral-75 hover:x-text-neutral-0 selected:x-text-neutral-0 selected:hover:x-bg-lead-50"
```
Comment (diegopf) [inline packages/x-components/src/x-modules/related-prompts/components/related-prompt.vue:17]: Same here. Probably is better to have a prop to pass the desired classes. This way is easier from the outside customize the UI

File: packages/x-components/src/views/home/Home.vue [hunk 3]
```diff
@@ -390,8 +390,9 @@
 
                           <template #related-prompts-group="{ item: { relatedPrompts } }">
                             <RelatedPrompt
-                              class="x-flex x-flex-col x-pt-24"
+                              class="x-bg-neutral-10 x-flex x-flex-col x-gap-16 x-pt-24"
```
Comment (annacv) [inline packages/x-components/src/views/home/Home.vue:393]: Hi, maybe this background is another possible customizable style for customers, maybe we could add a `containerClass` in the component to handle this attribute, which could also be used to add the gap & other spacing

File: packages/x-components/src/views/home/Home.vue [hunk 4]
```diff
@@ -390,8 +390,9 @@
...
                               :relatedPrompt="relatedPrompts[0]"
+                              nextQueryButtonClass="x-button x-button-lead x-button-sm x-button-outlined"
```
Comment (annacv) [inline packages/x-components/src/views/home/Home.vue:395]: Hi, IMO the `x-button` would rely on the component itself, the rest of them ok here

File: packages/x-components/src/x-modules/related-prompts/components/related-prompt.vue [hunk 5]
```diff
@@ -1,38 +1,26 @@
...
         <SlidingPanel :resetOnContentChange="false">
-          <div class="x-flex x-gap-8 x-pr-8">
+          <div class="x-flex x-gap-8">
             <button
               v-for="(nextQuery, index) in relatedPrompt.nextQueries"
               :key="index"
               @click="onClick(nextQuery)"
-              class="x-button x-button-lead x-button-sm x-button-outlined x-rounded-sm x-border-lead-50 x-text-neutral-75 hover:x-text-neutral-0 selected:x-text-neutral-0 selected:hover:x-bg-lead-50"
-              :class="{ 'x-selected': selectedNextQuery === nextQuery }"
+              :class="[{ 'x-selected': selectedNextQuery === nextQuery }, nextQueryButtonClass]"
```
Comment (annacv) [inline packages/x-components/src/x-modules/related-prompts/components/related-prompt.vue:14]: Hi, I would name it `buttonClass` so it is more agnostic about its content (who knows if they will finally be next queries or another kind of suggestion in the future)

File: packages/x-components/src/x-modules/related-prompts/components/related-prompt.vue [hunk 6]
```diff
@@ -1,38 +1,26 @@
...
-                class="x-whitespace-nowrap"
-                :class="
-                  selectedNextQuery === nextQuery ? 'x-title3 x-title3-md' : 'x-text1 x-text1-lg'
-                "
-              >
-                {{ nextQuery }}
-              </span>
+              <span>{{ nextQuery }}</span>
               <CrossTinyIcon v-if="selectedNextQuery === nextQuery" class="x-icon" />
               <PlusIcon v-else class="x-icon" />
```
Comment (annacv) [inline packages/x-components/src/x-modules/related-prompts/components/related-prompt.vue]: Maybe it is worth also adding a slot for the item

**Comment by annacv** [comment]
Hi @victorcg88! Great job! 
I've added some suggestions, & I miss the generic component description above the defineComponent f() in related-prompt.vue.

File: packages/x-components/src/x-modules/related-prompts/components/related-prompt.vue (hunk 5 above)
Comment (diegopf) [reply packages/x-components/src/x-modules/related-prompts/components/related-prompt.vue:14]: There are more than one button, I think it's better to have a more clear name for this

File: packages/x-components/src/views/home/Home.vue (hunk 3 above)
Comment (victorcg88) [reply packages/x-components/src/views/home/Home.vue:393]: You can already achieve this by attribute inheritance. Those classes will fall to the root element inside `RelatedPrompt`, just like is done here.

**Review by diegopf — APPROVED** [review]

**Review by annacv — APPROVED** [review]

**Pull request merged**